wasm2c/wasm-rt.h

```diff
@@ -166,17 +166,42 @@ extern "C" {
 #define WASM_RT_INSTALL_SIGNAL_HANDLER 0
 #endif
 
-#ifndef WASM_RT_USE_STACK_DEPTH_COUNT
-/* The signal handler on POSIX can detect call stack overflows. On windows, or
- * platforms without a signal handler, we use stack depth counting. */
+/* We need to detect and trap stack overflows. If we use a signal handler on
+ * POSIX systems, this can detect call stack overflows. On windows, or platforms
+ * without a signal handler, we use stack depth counting. */
+#if !defined(WASM_RT_STACK_DEPTH_COUNT) && \
+    !defined(WASM_RT_STACK_SIGNAL_HANDLER)
+
 #if WASM_RT_INSTALL_SIGNAL_HANDLER && !defined(_WIN32)
-#define WASM_RT_USE_STACK_DEPTH_COUNT 0
+#define WASM_RT_STACK_SIGNAL_HANDLER 1
```

Comment:
I'm not sure "stack signal handler" conveys that this is about handling a segfault signal triggered by stack _exhaustion_. How would you feel about `WASM_RT_STACK_EXHAUSTION_HANDLER` or something like that?